packages/vite/src/node/server/transformRequest.ts

```diff
@@ -419,37 +438,40 @@ async function loadAndTransform(
  * - SSR: We don't need to change anything as `ssrLoadModule` controls it
  */
 async function handleModuleSoftInvalidation(
-  mod: ModuleNode,
-  ssr: boolean,
+  environment: DevEnvironment,
+  mod: EnvironmentModuleNode,
   timestamp: number,
-  server: ViteDevServer,
 ) {
-  const transformResult = ssr ? mod.ssrInvalidationState : mod.invalidationState
+  const transformResult = mod.invalidationState
 
   // Reset invalidation state
-  if (ssr) mod.ssrInvalidationState = undefined
-  else mod.invalidationState = undefined
+  mod.invalidationState = undefined
 
   // Skip if not soft-invalidated
   if (!transformResult || transformResult === 'HARD_INVALIDATED') return
 
-  if (ssr ? mod.ssrTransformResult : mod.transformResult) {
+  if (mod.transformResult) {
     throw new Error(
       `Internal server error: Soft-invalidated module "${mod.url}" should not have existing transform result`,
     )
   }
 
   let result: TransformResult
-  // For SSR soft-invalidation, no transformation is needed
-  if (ssr) {
+  // No transformation is needed if it's disabled manually
+  // This is primarily for backwards compatible SSR
+  if (!environment.config.injectInvalidationTimestamp) {
```

Comment:
Why does SSR HMR now need to inject the invalidation timestamp to work? Aren't we keeping the module state internally so we can provide the updated module as needed?

Reply:
This was actually from an earlier idea to reuse the same logic the browser has, but now we do this in the transport layer instead

https://github.com/vitejs/vite/blob/8e368234937abfa63ee2d4d513876bd019d49097/packages/vite/src/module-runner/runner.ts#L289

It is not used for SSR anymore, we just set it by default if the consumer is `client`, but I think it's fine to keep this to have more granular control - I am also not against removing this completely since we don't have a use case inside Vite

Reply:
If we don't need this option at this point, we can remove it and have the condition by `environment.config.consumer === 'server'` as you propose. We can later add it when we have a good external (or internal) use case.

Reply:
Removed it https://github.com/vitejs/vite/pull/16471/commits/628c7a605a6c9cf95ac0b786d4da4dd55547d35f